src/app/modules/main/chat_section/module.nim

```diff
@@ -577,23 +589,41 @@ method onContactUnblocked*(self: Module, publicKey: string) =
   self.view.chatsModel().blockUnblockItemOrSubItemById(publicKey, blocked=false)
 
 method onContactDetailsUpdated*(self: Module, publicKey: string) =
-  let contact = self.controller.getContact(publicKey)
-  if (contact.requestReceived() and
-    not contact.isContact()and
-    not contact.isBlocked() and
+  let contactDetails = self.controller.getContactDetails(publicKey)
+  if (contactDetails.details.requestReceived() and
+    not contactDetails.details.isContact()and
+    not contactDetails.details.isBlocked() and
     not self.view.contactRequestsModel().containsItemWithPubKey(publicKey)):
       let item = self.createItemFromPublicKey(publicKey)
       self.view.contactRequestsModel().addItem(item)
+      self.updateParentBadgeNotifications()
+      singletonInstance.globalEvents.showNewContactRequestNotification("New Contact Request", 
+      fmt "{contactDetails.displayName} added you as contact", conf.CHAT_SECTION_ID) 
 
-  let (chatName, chatImage, isIdenticon) = self.controller.getOneToOneChatNameAndImage(publicKey)
+  let chatName = contactDetails.displayName
+  let chatImage = contactDetails.icon
+  let isIdenticon = contactDetails.isIdenticon
   self.view.chatsModel().updateItemDetails(publicKey, chatName, chatImage, isIdenticon)
 
 method onNewMessagesReceived*(self: Module, chatId: string, unviewedMessagesCount: int, unviewedMentionsCount: int,
   messages: seq[MessageDto]) =
   if(self.controller.getMySectionId() == self.delegate.getActiveSectionId() and
     self.controller.getActiveChatId() == chatId):
     return
-  self.updateNotifications(chatId, unviewedMessagesCount, unviewedMentionsCount)
+  self.updateBadgeNotifications(chatId, unviewedMessagesCount, unviewedMentionsCount)
+
+  # Prepare bubble notification
+  let myPK = singletonInstance.userProfile.getPubKey()
+  for m in messages:
+    let contactDetails = self.controller.getContactDetails(m.`from`)
+    let renderedMessageText = self.controller.getRenderedText(m.parsedText)
+    let plainText = singletonInstance.utils.plainText(renderedMessageText)
+    if(m.isUserWithPkMentioned(myPK)):
+      singletonInstance.globalEvents.showMentionMessageNotification(contactDetails.displayName, plainText, 
+      self.controller.getMySectionId(), chatId, m.id)
+    else:
+      singletonInstance.globalEvents.showNormalMessageNotification(contactDetails.displayName, plainText, 
```

Comment:
`showMentionMessageNotification` and `showMentionMessageNotification` and really similar, maybe we can merge them and just add a flag `isMention`?

Reply:
@jrainville that's other way to proceed and use `type` param for example to differ them, but I did it this way because of handling in `NoitificationManager` class. The effect at the end should be the same, but this way we don't need to include file where enum type is defined cause I wouldn't like to add some magic numbers like 1, 2, 3 for types.